### PR TITLE
Orb gets SAD

### PR DESCRIPTION
**Pull request by Moth-lantern**
## About The Pull Request
Okay so stop me if this has happened to you, you're havin a good shift doin your best and suddenly your workplace explodes and you're reduced to just a head, Kind of a bummer right? Well anyway then the doctor goes to fix you and there aren't any matching bodies to put your head on and you end up just havin to be borged or deal with dysphoria! WELL worry no longer! with the help of Interdyne pharmaceuticals we've made the Self Actualization Device, a safe and completely tested (on monkey's) machine allowing you to become your true self once again!
whoag it's the Sad

I am being scanned

## Why It's Good For The Game
We've had a lot of issues with people getting dysphoria from head transplants so now? worry no longer!

## Changelog

:cl:
add: ports the S.A.D from skyrat
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_defines.dm [hunk 1]
```diff
@@ -15,6 +15,10 @@
 	var/list/obj/item/organ/internal/organs = list()
 	///Same as [above][/mob/living/carbon/var/organs], but stores "slot ID" - "organ" pairs for easy access.
 	var/list/organs_slot = list()
+	///List of [/obj/item/organ/external] in the mob, similarly used as internal_organs.
+	var/list/obj/item/organ/external/external_organs = list()
+	///Same as [above][/mob/living/carbon/var/external_organs], but stores "ID" = "organ" pairs.
+	var/list/external_organs_slot = list()
```
Comment (flowercuco) [inline code/modules/mob/living/carbon/carbon_defines.dm:21]: mark these as orbstation changes in the comments somewhere

**Review by flowercuco — CHANGES_REQUESTED** [review]
just have to be explicit when doing changes to upstream files but otherwise i think this is fine?

File: code/modules/mob/living/carbon/carbon_defines.dm (hunk 1 above)
Comment (Moth-lantern) [reply code/modules/mob/living/carbon/carbon_defines.dm:21]: agh I had commented these and my pc powered off and I forgot to do this again

**Review by flowercuco — APPROVED** [review]

File: orbstation/code/objects/machinery/SAD.dm [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+/obj/machinery/self_actualization_device/open_machine(drop = TRUE, density_to_set = FALSE)
+	playsound(src, 'sound/machines/click.ogg', 50)
+	..()
+
+/obj/machinery/self_actualization_device/AltClick(mob/user)
+	. = ..()
+	if(!powered() || !occupant || state_open)
+		return FALSE
+
+	to_chat(user, "You power on [src].")
```
Comment (Jacquerel) [inline orbstation/code/objects/machinery/SAD.dm:127]: I'd like these more as balloon alerts but it's not very important.

File: orbstation/code/objects/machinery/SAD.dm [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+		return FALSE
+
+	to_chat(user, span_notice("The emergency release is not responding! You start pushing against the hull!"))
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	user.visible_message(span_notice("You see [user] kicking against the door of [src]!"), \
+		span_notice("You lean on the back of [src] and start pushing the door open... (this will take about [DisplayTimeText(breakout_time)].)"), \
+		span_hear("You hear a metallic creaking from [src]."))
+
+	if(do_after(user, breakout_time, target = src))
```
Comment (Jacquerel) [inline orbstation/code/objects/machinery/SAD.dm:144]:
```suggestion
	if(!do_after(user, breakout_time, target = src))
		return
```

File: orbstation/code/objects/machinery/SAD.dm [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+	next_fact--
+	if(next_fact <= 0)
+		next_fact = rand(initial(next_fact), 2 * initial(next_fact))
+		say(pick(advertisements))
+		playsound(loc, 'sound/machines/chime.ogg', 30, FALSE)
+
+	use_power(500)
+
+/// Ejects the occupant as either their preference character, or as a monke based on emag status.
```
Comment (Jacquerel) [inline orbstation/code/objects/machinery/SAD.dm:175]: This comment does not seem to be true

Comment (Moth-lantern) [reply orbstation/code/objects/machinery/SAD.dm:175]: sorry some of this was just ported directly from skyrat code and it seems thats just a goof comment they left on the S.A.D even though it doesn't actually seem to do that.

**Review by Jacquerel — APPROVED** [review]

**Pull request merged**